src/render/draw_fill_extrusion.js

```diff
@@ -70,9 +70,6 @@ function drawExtrusionTiles(painter, source, layer, coords, depthMode, stencilMo
     const baseDefines = ([]: any);
     if (isGlobeProjection) {
         baseDefines.push('PROJECTION_GLOBE_VIEW');
-        if (painter.style.terrainSetForDrapingOnly()) {
-            baseDefines.push('TERRAIN');
-        }
```

Comment:
Do you recall the need for this diff?

Reply:
It is about this:

> An issue with globe fill extrusion (with terrain disabled) got exposed in the bot: when TERRAIN is defined, centroid is used in vertex shader. However, in that case, we don't call gl.vertexAttrib2f(program.attributes['a_centroid_pos'], 0, 0), and this leads to undefined vertex attrib data (e.g. zero-height-clipping-angle-limit test). Globe 2D shaders don't seem to depend on TERRAIN define, so the original code setting the define is removed.

defined TERRAIN activates vertex shader path:

https://github.com/mapbox/mapbox-gl-js/blob/67ed1ab99a9ffe9f0f9266ad0ca3ef438d8faf89/src/shaders/fill_extrusion.vertex.glsl#L80-L83